sphinx_gallery/notebook.py

```diff
@@ -142,36 +145,70 @@ def add_code_cell(work_notebook, code):
     work_notebook["cells"].append(code_cell)
 
 
-def add_markdown_cell(work_notebook, text):
+def add_markdown_cell(work_notebook, text, pandoc=False):
     """Add a markdown cell to the notebook
 
     Parameters
     ----------
     code : str
         Cell content
     """
-    markdown_cell = {
-        "cell_type": "markdown",
-        "metadata": {},
-        "source": [rst2md(text)]
-    }
+    if pandoc:
+        markdown_cell = {
+            "cell_type": "markdown",
+            "metadata": {},
+            "source": [text]
+        }
+    else:
+        markdown_cell = {
+            "cell_type": "markdown",
+            "metadata": {},
+            "source": [rst2md(text)]
+        }
     work_notebook["cells"].append(markdown_cell)
 
 
-def fill_notebook(work_notebook, script_blocks):
+def fill_notebook(work_notebook, script_blocks, gallery_conf):
     """Writes the Jupyter notebook cells
 
+    If available, uses pypandoc to convert rst to markdown.
+
     Parameters
     ----------
     script_blocks : list
         Each list element should be a tuple of (label, content, lineno).
     """
+    pandoc = False
+    pandoc_kwargs = {}
+    if gallery_conf["pypandoc"] or isinstance(gallery_conf["pypandoc"], dict):
+        try:
+            import pypandoc  # noqa
+            # Import error raised only when function called
+            logger.info("pandoc version: %s"
+                        % (pypandoc.get_pandoc_version(),))
+        except (ImportError, OSError) as e:
+            logger.warning("'pypandoc' not available. Using Sphinx-Gallery to "
+                           "convert rst text blocks to markdown for .ipynb "
+                           "files.")
+            if e:
+                logger.warning("pypandoc import error: %s" % (e,))
+        else:
+            pandoc = True
+            if isinstance(gallery_conf["pypandoc"], dict):
+                pandoc_kwargs = gallery_conf["pypandoc"]
 
     for blabel, bcontent, lineno in script_blocks:
         if blabel == 'code':
             add_code_cell(work_notebook, bcontent)
         else:
-            add_markdown_cell(work_notebook, bcontent + '\n')
+            if pandoc:
+                md = pypandoc.convert_text(
+                    bcontent, to='md', format='rst', **pandoc_kwargs
+                )
+                # pandoc automatically adds '\n' at end
+                add_markdown_cell(work_notebook, md, pandoc=True)
+            else:
+                add_markdown_cell(work_notebook, bcontent + '\n')
 
```

Comment:
Alternatively, maybe you could do above:
```
converter = partial(pypandoc.convert_text, to='md', format='rst', **pandoc_kwargs)  # if pandoc
...
converter = None  # if not pandoc
```
And then this call can just be:
```
add_markdown_cell(work_notebook, bcontent + '\n', converter=converter)
```
And `add_markdown_cell` has `converter = rst2md if converter is None else converter`.

To me it seems to have a nicer logic/flow compared to the current implementation, WDYT?

Reply:
(and maybe we just kill `add_markdown_cell` entirely and keep the logic all in this function, since now `add_markdown_cell` does almost nothing)

Reply:
ahh but I see there is a `add_code_cell`, so probably worth keeping `add_markdown_cell`. But maybe we should give it content already in MD format, rather than asking it to do the conversion